assets/javascripts/discourse/components/modal/post-event-builder.js

```diff
@@ -0,0 +1,202 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { extractError } from "discourse/lib/ajax-error";
+import { cook } from "discourse/lib/text";
+import Group from "discourse/models/group";
+import { buildParams, replaceRaw } from "../../lib/raw-event-helper";
+
+export default class PostEventBuilder extends Component {
+  @service dialog;
+  @service siteSettings;
+  @service store;
+
+  @tracked flash = null;
+  @tracked startsAt = moment(this.args.model.event.starts_at).tz(
+    this.args.model.event.timezone || "UTC"
+  );
+  @tracked
+  endsAt =
+    this.args.model.event.ends_at &&
+    moment(this.args.model.event.ends_at).tz(
+      this.args.model.event.timezone || "UTC"
+    );
+
+  get reminderTypes() {
+    return [
+      { value: "notification", name: "Notification" },
+      { value: "bumpTopic", name: "Bump Topic" },
+    ];
+  }
+
+  get reminderUnits() {
+    return [
+      { value: "minutes", name: "Minutes" },
+      { value: "hours", name: "Hours" },
+      { value: "days", name: "Days" },
+      { value: "weeks", name: "Weeks" },
+    ];
+  }
+
+  get reminderPeriods() {
+    return [
+      { value: "before", name: "Before" },
+      { value: "after", name: "After" },
+    ];
+  }
+
+  get availableRecurrences() {
+    return [
+      { id: "every_day", name: "Every Day" },
+      { id: "every_month", name: "Every Month" },
+      { id: "every_weekday", name: "Every Weekday" },
+      { id: "every_week", name: "Every Week" },
+      { id: "every_two_weeks", name: "Every Two Weeks" },
+      { id: "every_four_weeks", name: "Every Four Weeks" },
+    ];
+  }
+
+  get allowedCustomFields() {
+    return this.siteSettings.discourse_post_event_allowed_custom_fields
+      .split("|")
+      .filter(Boolean);
+  }
+
+  get addReminderDisabled() {
+    return this.args.model.event.reminders?.length >= 5;
+  }
+
+  @action
+  groupFinder(term) {
+    return Group.findAll({ term, ignore_automatic: true });
+  }
+
+  @action
+  setCustomField(field, target) {
+    this.args.model.updateCustomField(field, target.value);
+  }
+
+  @action
+  onChangeDates(dates) {
+    this.args.model.onChangeDates(dates);
+    this.startsAt = dates.from;
+    this.endsAt = dates.to;
+  }
+
+  @action
+  onChangeStatus(newStatus) {
+    this.args.model.updateEventRawInvitees([]);
+    if (newStatus === "private") {
+      this.args.model.updateEventRawInvitees(
+        this.args.model.event.raw_invitees.filter((x) => x !== "trust_level_0")
+      );
+    }
+    this.args.model.updateEventStatus(newStatus);
```

Comment:
This is basically copy pasted, and it doesn't make sense imo. I think it could be something like

```suggestion
    this.args.model.updateEventRawInvitees([]);
    this.args.model.updateEventStatus(newStatus);
```